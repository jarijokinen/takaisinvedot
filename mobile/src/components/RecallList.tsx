import React, { useEffect, useState } from 'react';
import { FlatList, Linking, TouchableOpacity } from 'react-native';
import { ListItem } from 'react-native-elements';

import firestore from '@react-native-firebase/firestore';
import { firestore as firebaseFirestore } from 'firebase';

export interface IRecallItem {
  id: string;
  title: string;
  url: string;
  date: firebaseFirestore.Timestamp;
  source: string;
}

const RecallList: React.FC = () => {
  const [recalls, setRecalls] = useState<Array<IRecallItem> | undefined>([]);

  useEffect(() => {
    firestore()
      .collection('recalls')
      .orderBy('date', 'desc')
      .limit(30)
      .onSnapshot((querySnapshot) => {
        const recallList: Array<IRecallItem> = [];

        querySnapshot.forEach((documentSnapshot) => {
          const data: IRecallItem = documentSnapshot.data() as IRecallItem;
          data.id = documentSnapshot.id;
          recallList.push(data);
        });

        setRecalls(recallList);
      });
  }, []);

  const onPressItem = (item: IRecallItem) => {
    Linking.openURL(item.url);
  };

  const formatDate = (date: Date): string => {
    return [date.getDate(), date.getMonth() + 1, date.getFullYear()].join('.');
  };

  const itemIcon = (item: IRecallItem) => {
    if (item.source === 'Ruokavirasto') {
      return { name: 'silverware-fork-knife', type: 'material-community' };
    } else {
      if (item.title.match(/Sähköisku/)) {
        return { name: 'flash', type: 'material-community' };
      } else if (item.title.match(/Palo/)) {
        return { name: 'fire', type: 'material-community' };
      } else if (item.title.match(/Puristuminen/)) {
        return { name: 'compress-arrows-alt', type: 'font-awesome-5' };
      } else if (item.title.match(/Kemiallinen/)) {
        return { name: 'chemical-weapon', type: 'material-community' };
      } else {
        return { name: 'warning', type: 'material' };
      }
    }
  };

  return (
    <FlatList
      data={recalls}
      renderItem={({ item }) => (
        <TouchableOpacity onPress={() => onPressItem(item)}>
          <ListItem
            title={item.title}
            subtitle={item.source + ' - ' + formatDate(item.date.toDate())}
            leftIcon={itemIcon(item)}
            bottomDivider
          />
        </TouchableOpacity>
      )}
    />
  );
};

export default RecallList;
