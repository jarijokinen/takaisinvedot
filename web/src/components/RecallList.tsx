import { useCollection } from '@nandorojo/swr-firestore';
import { firestore } from 'firebase';
import { formatDate } from '../utils/formatDate';

import * as Styled from './RecallList.styled';

export interface IRecallItem {
  id?: string;
  title?: string;
  url?: string;
  date?: firestore.Timestamp;
  source?: string;
}

const RecallList: React.FC = () => {
  const { data, error } = useCollection('recalls', {
    orderBy: ['date', 'desc'],
    limit: 30
  });

  if (error) return <div>Virhe tietoja ladatessa...</div>;
  if (!data) return <div>Ladataan...</div>;

  return (
    <Styled.RecallList>
      {data.map((item: IRecallItem) => (
        <Styled.Item key={item.id}>
          <Styled.ItemDate>{formatDate(item.date.toDate())}</Styled.ItemDate>
          <Styled.ItemLink href={item.url}>{item.title}</Styled.ItemLink>
          <Styled.ItemSource>{item.source}</Styled.ItemSource>
        </Styled.Item>
      ))}
    </Styled.RecallList>
  );
};

export default RecallList;
