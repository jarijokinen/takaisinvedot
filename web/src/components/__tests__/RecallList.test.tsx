import '@testing-library/jest-dom/extend-expect';

import { render } from '@testing-library/react';
import { firestore } from 'firebase';
import { useCollection } from '@nandorojo/swr-firestore';

import RecallList, { IRecallItem } from '../RecallList';

import { formatDate } from '../../utils/formatDate';

jest.mock('@nandorojo/swr-firestore');

const buildRecallItem = (id: number, overrides?: IRecallItem): IRecallItem => ({
  id: 'abc' + id,
  title: 'Example Title ' + id,
  url: 'http://example.com/example-path-' + id,
  date: firestore.Timestamp.fromDate(new Date(2020, 5, 20)),
  source: 'Example Source ' + id,
  ...overrides
});

const buildData = (count: number): Array<IRecallItem> => {
  const arr = [];
  for (let i = 0; i < count; i++) {
    arr.push(buildRecallItem(i + 1));
  }
  return arr;
};

let data = [];
let error = false;

describe('RecallList', () => {
  beforeAll(() => {
    data = buildData(50);
    error = false;
    (useCollection as jest.Mock).mockReturnValue({ data, error });
  });

  it('renders correctly', () => {
    const { asFragment } = render(<RecallList />);
    expect(asFragment()).toMatchSnapshot();
  });

  describe('item', () => {
    it('has correct title', () => {
      const { getByText } = render(<RecallList />);
      expect(getByText(data[0].title)).toBeInTheDocument();
    });

    it('has correct url', () => {
      const { getByText } = render(<RecallList />);
      expect(getByText(data[0].title)).toHaveAttribute('href', data[0].url);
    });

    it('has correct date', () => {
      const { container } = render(<RecallList />);
      expect(container.firstChild).toHaveTextContent(
        formatDate(data[0].date.toDate())
      );
    });

    it('has correct source', () => {
      const { container } = render(<RecallList />);
      expect(container.firstChild).toHaveTextContent(data[0].source);
    });
  });

  describe('with error', () => {
    beforeAll(() => {
      data = [];
      error = true;
      (useCollection as jest.Mock).mockReturnValue({ data, error });
    });

    it('renders error message', () => {
      const { getByText } = render(<RecallList />);
      expect(getByText('Virhe tietoja ladatessa...')).toBeInTheDocument();
    });
  });

  describe('with non-existing data', () => {
    beforeAll(() => {
      data = null;
      error = false;
      (useCollection as jest.Mock).mockReturnValue({ data, error });
    });

    it('renders loading message', () => {
      const { getByText } = render(<RecallList />);
      expect(getByText('Ladataan...')).toBeInTheDocument();
    });
  });
});
