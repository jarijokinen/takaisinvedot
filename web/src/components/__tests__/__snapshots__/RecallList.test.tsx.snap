// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecallList renders correctly 1`] = `
<DocumentFragment>
  <ul
    class="sc-AxjAm bfZAWj"
  >
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-1"
      >
        Example Title 1
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 1
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-2"
      >
        Example Title 2
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 2
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-3"
      >
        Example Title 3
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 3
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-4"
      >
        Example Title 4
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 4
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-5"
      >
        Example Title 5
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 5
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-6"
      >
        Example Title 6
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 6
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-7"
      >
        Example Title 7
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 7
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-8"
      >
        Example Title 8
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 8
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-9"
      >
        Example Title 9
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 9
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-10"
      >
        Example Title 10
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 10
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-11"
      >
        Example Title 11
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 11
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-12"
      >
        Example Title 12
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 12
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-13"
      >
        Example Title 13
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 13
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-14"
      >
        Example Title 14
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 14
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-15"
      >
        Example Title 15
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 15
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-16"
      >
        Example Title 16
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 16
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-17"
      >
        Example Title 17
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 17
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-18"
      >
        Example Title 18
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 18
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-19"
      >
        Example Title 19
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 19
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-20"
      >
        Example Title 20
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 20
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-21"
      >
        Example Title 21
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 21
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-22"
      >
        Example Title 22
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 22
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-23"
      >
        Example Title 23
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 23
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-24"
      >
        Example Title 24
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 24
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-25"
      >
        Example Title 25
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 25
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-26"
      >
        Example Title 26
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 26
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-27"
      >
        Example Title 27
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 27
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-28"
      >
        Example Title 28
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 28
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-29"
      >
        Example Title 29
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 29
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-30"
      >
        Example Title 30
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 30
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-31"
      >
        Example Title 31
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 31
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-32"
      >
        Example Title 32
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 32
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-33"
      >
        Example Title 33
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 33
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-34"
      >
        Example Title 34
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 34
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-35"
      >
        Example Title 35
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 35
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-36"
      >
        Example Title 36
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 36
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-37"
      >
        Example Title 37
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 37
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-38"
      >
        Example Title 38
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 38
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-39"
      >
        Example Title 39
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 39
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-40"
      >
        Example Title 40
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 40
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-41"
      >
        Example Title 41
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 41
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-42"
      >
        Example Title 42
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 42
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-43"
      >
        Example Title 43
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 43
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-44"
      >
        Example Title 44
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 44
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-45"
      >
        Example Title 45
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 45
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-46"
      >
        Example Title 46
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 46
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-47"
      >
        Example Title 47
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 47
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-48"
      >
        Example Title 48
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 48
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-49"
      >
        Example Title 49
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 49
      </div>
    </li>
    <li
      class="sc-AxirZ dNueBK"
    >
      <div
        class="sc-AxhCb vyKjO"
      >
        20.6.2020
      </div>
      <a
        class="sc-AxiKw exSGiu"
        href="http://example.com/example-path-50"
      >
        Example Title 50
      </a>
      <div
        class="sc-AxhUy cYQArD"
      >
        Example Source 50
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
